src/focus_manager/txm_platform.js

```diff
@@ -570,5 +581,62 @@ export class TXMPlatform {
             actionKeyCodes[inputActions.nextTrack] = keyCodes.O;
         }
     }
+
+    /**
+     * Returns a promise that resolves to the user's advertising id for the platform. Resolves to
+     * undefined if the advertising id is either not available or else the user has opted out of
+     * being tracked for advertising on their device.
+     *
+     * @return {Promise<String>}
+     */
+    async getUserAdvertisingId() {
+        if (!this.supportsUserAdvertisingId) {
+            return undefined;
+        }
+
+        if (this.isFireTV) {
+            return this.getFireTVAdvertisingId();
+        }
+
+        return undefined; // fallback
+    }
+
+    async getFireTVAdvertisingId() {
+        let AmazonAdvertising = window.AmazonAdvertising;
+        if (!AmazonAdvertising) {
+            const apiLoader = new ScriptLoader("https://resources.amazonwebapps.com/v1/latest/Amazon-Web-App-API.min.js");
+            apiLoader.load();
+            await apiLoader.promise;
+            AmazonAdvertising = await new Promise(resolve => {
+                document.addEventListener('amazonPlatformReady', onApiReady);
+
+                function onApiReady() {
+                    document.removeEventListener('amazonPlatformReady', onApiReady);
+                    resolve(window.AmazonAdvertising);
+                }
+            });
+            if (!AmazonAdvertising) {
+                throw new Error("AmazonAdvertising API not available");
+            }
+        }
+        const adIdPromise = new Promise((resolve, reject) => {
+            AmazonAdvertising.getAdvertisingId(resolve, errMsg => {
+                console.error(`getAdvertisingId: ${errMsg}`);
+                resolve(undefined);
+            })
+        });
+        const adTrackingPromise = new Promise((resolve, reject) => {
+            AmazonAdvertising.getLimitAdTrackingPreference(resolve, errMsg => {
+                console.error(`getLimitAdTrackingPreference: ${errMsg}`);
+                resolve(false);
+            })
+        });
```

Comment:
I had to look up the AmazonAdvertising documentation to understand this part. Not really sure if there's a better way to write it though..

Reply:
It's pretty much as good as i can think of to do the requests in parallel